src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java

```diff
@@ -39,4 +63,94 @@ public WarehouseDto save(WarehouseDto dto) {
+        return Objects.isNull(availableCapacity) ?
+                warehouseRepository.getTotalCapacity(warehouseId)
+                : availableCapacity;
+    }
+
+    private Set<ItemsInApplicationDto> getItemsToAccept(final Set<ItemsInApplicationDto> itemToAccept,
+                                                        final Long appId) {
+        final List<Long> acceptedItemInAppId = itemToAccept
+                .stream()
+                .map(ItemsInApplicationDto::getId)
+                .collect(Collectors.toList());
+        final Set<ItemsInApplicationDto> acceptedItemFromDb = applicationService
+                .getItemsById(acceptedItemInAppId, appId);
+        if (acceptedItemFromDb.size() != itemToAccept.size()) {
+            throw new ConflictWithTheCurrentStateException("Attempt to accept doesn't exist or already accepted item");
+        }
+        return acceptedItemFromDb;
+    }
+
+    private void acceptToWarehouse(final Set<ItemsInApplicationDto> acceptedToWarehouse,
+                                   final Long warehouseId) {
+        final List<Long> acceptedIds = new ArrayList<>();
+        acceptedToWarehouse.forEach(itemToAccept -> {
+            final Long itemId = itemToAccept.getItemDto().getId();
+            final Optional<ItemsInWarehouse> iiw = itemInWarehouseRepository.findByItemId(itemId, warehouseId);
+            final ItemsInWarehouse itemsInWarehouse = iiw.map(
+                    items -> updateItemInWarehouse(items, itemToAccept))
+                    .orElseGet(() -> createItemInWarehouse(warehouseId, itemToAccept));
+            acceptedIds.add(itemToAccept.getId());
+            itemInWarehouseRepository.save(itemsInWarehouse);
+        });
+        applicationService.setItemInApplicationAcceptedAt(acceptedIds);
+    }
+
+    private ItemsInWarehouse updateItemInWarehouse(final ItemsInWarehouse itemsInWarehouse,
+                                                   final ItemsInApplicationDto itemsInApplication) {
+        itemsInWarehouse.setAmount(itemsInWarehouse.getAmount() + itemsInApplication.getAmount());
+        itemsInWarehouse.setCost(
+                itemsInWarehouse.getCost().compareTo(itemsInApplication.getCost()) >= 0 ?
+                        itemsInWarehouse.getCost() :
+                        itemsInApplication.getCost());
+        return itemsInWarehouse;
+    }
+
+    private ItemsInWarehouse createItemInWarehouse(final Long warehouseId,
+                                                   final ItemsInApplicationDto itemsInApplication) {
+        return ItemsInWarehouse.builder()
+                .item(itemMapper.map(itemsInApplication.getItemDto()))
+                .amount(itemsInApplication.getAmount())
+                .cost(itemsInApplication.getCost())
+                .warehouse(Warehouse.builder().id(warehouseId).build())
+                .build();
+    }
+
+    private void checkWarehouseCapacity(final Set<ItemsInApplicationDto> acceptedToWarehouse,
```

Comment:
I think that throwing exception from this method is not very good approach. If we use this method it is not obvious that it throws an exception in some case. And what does it do it not an exceptional case? We could return a boolean result here and in the place where this method is used throw an exception. 
@vareyko-itechart , what do you think of it?

Reply:
in a common case - yes, but as I understood its causes interruption of the whole action and there is a lot of checks. I would suggest the refactoring of the original method to have all checks in the same place and handle it with some response (maybe with some specific details), not only the exception